examples/prom/maxwell_global_rom.cpp

```diff
@@ -0,0 +1,548 @@
+            generator->writeSnapshot();
+            delete generator;
+            delete options;
+        }
+    }
+
+    // 19. The online phase
+    if (online) {
+        // 20. read the reduced basis
+        assembleTimer.Start();
+        CAROM::BasisReader reader(basisName);
+        spatialbasis = reader.getSpatialBasis(0.0);
+        numRowRB = spatialbasis->numRows();
+        numColumnRB = spatialbasis->numColumns();
+        if (myid == 0) printf("spatial basis dimension is %d x %d\n", numRowRB,
+                                  numColumnRB);
+
+        // libROM stores the matrix row-wise, so wrapping as a DenseMatrix in MFEM means it is transposed.
+        DenseMatrix *reducedBasisT = new DenseMatrix(spatialbasis->getData(),
+                numColumnRB, numRowRB);
+
+        // 21. form inverse ROM operator
+        CAROM::Matrix invReducedA(numColumnRB, numColumnRB, false);
+        ComputeCtAB( *A.As<HypreParMatrix>(), *spatialbasis, *spatialbasis,
+                     invReducedA);
+        invReducedA.inverse();
+
+        CAROM::Vector B_carom(B.GetData(), B.Size(), true, false);
+        CAROM::Vector X_carom(X.GetData(), X.Size(), true, false);
+        CAROM::Vector *reducedRHS = spatialbasis->transposeMult(&B_carom);
+        CAROM::Vector reducedSol(numColumnRB, false);
+        assembleTimer.Stop();
+
+        // 22. solve ROM
+        solveTimer.Start();
+        invReducedA.mult(*reducedRHS, reducedSol);
+        solveTimer.Stop();
+
+        // 23. reconstruct FOM state
+        spatialbasis->mult(reducedSol, X_carom);
+        delete spatialbasis;
+        delete reducedRHS;
+    }
+
+    // 24. Recover the parallel grid function corresponding to X. This is the
+    //     local finite element solution on each processor.
+    a->RecoverFEMSolution(X, *b, x);
+
+    // 26. Calculate the relative error of the ROM prediction compared to FOM
+    ostringstream sol_dofs_name, sol_dofs_name_fom;
+    if (fom || offline)
+    {
+        sol_dofs_name << "sol_dofs_fom." << setfill('0') << setw(6) << myid;
+    }
+    if (online)
+    {
+        sol_dofs_name << "sol_dofs." << setfill('0') << setw(6) << myid;
+        sol_dofs_name_fom << "sol_dofs_fom." << setfill('0') << setw(6) << myid;
+    }
+
+    if (online)
+    {
+        // Initialize FOM solution
+        Vector x_fom(x.Size());
+
+        ifstream fom_file;
+
+        // Open and load file
+        fom_file.open(sol_dofs_name_fom.str().c_str());
+
+        x_fom.Load(fom_file, x_fom.Size());
+
+        fom_file.close();
+
+        Vector diff_x(x.Size());
+
+        subtract(x, x_fom, diff_x);
+
+        // Get norms
+        double tot_diff_norm = sqrt(InnerProduct(MPI_COMM_WORLD, diff_x, diff_x));
+
+        double tot_fom_norm = sqrt(InnerProduct(MPI_COMM_WORLD,
+                                                x_fom, x_fom));
+
+        if (myid == 0)
+        {
+            cout << "Relative error of ROM solution = "
```

Comment:
The relative error is over 3 (300%), although the solutions look very similar in all cases. It looks like the solution is reasonably accurate, but the error computation is wrong? Have you looked into this?

Reply:
Thanks for the comment. I realized the files being used to measure the relative error,  "sol_dofs...." and "sol_dofs_fom...",  are not even being defined. Not sure how this part was even running because of that.

Adding some lines to  part 27 to fix this. Relative error is now 0.000445366 for the example at the top of this script